60_networks.Rmd

```diff
@@ -0,0 +1,855 @@
+
+**NetCoMi package:** All of the above steps are also implemented in the R package [NetCoMi](https://github.com/stefpeschel/NetCoMi) [@peschel2021netcomi], which provides functionality for the complete workflow of constructing, analyzing, and comparing networks for microbiome data. A large part of this section, in particular the visualization and comparison of networks, is carried out using NetCoMi.
+
+## Load packages and data
+
+We start by installing `NetCoMi` from GitHub and loading other necessary packages used in this section.
+
+```{r install-pkg, eval=FALSE}
+if(!require(NetCoMi)){
+  devtools::install_github("stefpeschel/NetCoMi", force = TRUE,
+                         dependencies = c("Depends", "Imports", "LinkingTo"),
+                         repos = c("https://cloud.r-project.org/",
+                                   BiocManager::repositories()))
+}
+```
+
+```{r load_packages, message=FALSE, warning=FALSE}
+library(NetCoMi)
+library(igraph)
+library(mia)
+library(phyloseq)
+library(corpcor)
+library(pheatmap)
+library(grid)
+library(gridExtra)
+library(RColorBrewer)
+```
+
+The PeerJ data set [@potbhare2022skin] is used in this section. It contains skin microbial profiles of 58 subjects.
+
+```{r load_data}
+data("peerj13075", package = "mia")
+
+tse0 <- peerj13075
+```
+
+```{r, echo=FALSE}
+load("general/network_data/networks.RData")
+```
+
+## Network learning
+
+### Prevalence Filtering
+
+Before we apply the network learning methods, we perform some preprocessing steps similar to that in section \@ref(differential-abundance):
+
+-   Aggregation to genus level
+-   Add relative abundance assay
+-   Prevalence filtering (keep genera with prevalence \> 20%)
+-   Add assay with clr-transformed abundances
+
+```{r}
+# Agglomerate to genus level
+tse_genus <- agglomerateByRank(tse0, rank = "genus") 
+
+# Add relative abundances
+#debugonce(vegan::decostand)
+tse <- transformAssay(tse_genus,
+                       assay.type = "counts",
+                       method = "relabundance",
+                       MARGIN = "samples") 
+
+# Filter by prevalence
+tse <- subsetByPrevalentFeatures(tse,
+                                 detection = 0,
+                                 prevalence = 0.2,
+                                 assay.type = "relabundance")
+
+
+# Add clr-transformed abundances
+tse <- transformAssay(tse, method="clr", pseudocount=1)
+
+dim(tse0)
+dim(tse)
+```
+
+### SparCC correlation
+
+We start by estimating a sparse correlation matrix using SparCC.
+
+```{r, eval=FALSE}
+sparcc_cor <- SpiecEasi::sparcc(t(assay(tse, "counts")))$Cor
+rownames(sparcc_cor) <- colnames(sparcc_cor) <- rownames(tse)
+```
+
+Two threshold values are used to see the effect of sparsification later in the network plot.
+
+```{r}
+sparcc_cor03 <- sparcc_cor
+
+# Sparsification (threshold = 0.3)
+sparcc_cor03[abs(sparcc_cor03) < 0.3] <- 0
+
+# Compute dissimilarity matrix
+sparcc_diss03 <- sqrt(0.5 * (1 - sparcc_cor03))
```

Comment:
In fact, this approach is not standard. We propose it in the NetCoMi paper (Peschel et al, 2021) and want to make people aware of the differences in how negative associations are handled. Often, the absolute value of estimated associations is taken, resulting in high edge weights between highly negatively associated nodes. However, from a biological point of view, it also makes sense for negatively associated species to be placed far apart from each other in the network. We have included these two transformations in our workflow to allow users to decide how to handle negative associations. In our opinion, this step is crucial because it affects all distance-based network measures. However, it is often not even mentioned how associations are transformed into edge weights.

The transformations are explained in the introduction, but I will add a comment here as well.